lib/vector-core/src/metrics/handle.rs

```diff
@@ -59,9 +59,9 @@ impl Handle {
         Handle::Gauge(Gauge::new())
     }
 
-    pub(crate) fn update_gauge(&mut self, value: GaugeValue) {
+    pub(crate) fn update_gauge(&mut self, value: &GaugeValue) {
         match self {
-            Handle::Gauge(gauge) => gauge.record(value),
+            Handle::Gauge(gauge) => gauge.record(&value),
```

Comment:
Hmm, I know why clippy is asking for this, but this isn't really a win. `GaugeValue` can be represented in two words, which other lints would tell you to do the opposite to here. However, this appears to be be caused by a lack of `Copy`, which would be trivial on `GaugeValue`.

Reply:
This wins by a word, no? https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=40541b416ec1fbf58dad098273036ebd It's _very_ marginal but I assume that's what the lint is keying off of.

Reply:
While the reference is indeed just a single word, Rust knows enough to pass an two word struct/enum like that in registers, which avoids a potential memory spill and dereference (assuming it isn't inlined, of course).

Which lint actually triggered those changes?

Reply:
> While the reference is indeed just a single word, Rust knows enough to pass an two word struct/enum like that in registers, which avoids a potential memory spill and dereference (assuming it isn't inlined, of course).

TIL 

> Which lint actually triggered those changes?

Couldn't tell you directly but I know it's something from `pedantic` as this comes prior to 72afbbf which only flipped on `clippy::all`.

Reply:
Resolved in 7057547